### PR TITLE
Update to FPDI 2

### LINKED ISSUE #578: Feature: setasign/fpdi: 2.0 support
Please add support to setasign/fpdi: 2.0 version. 
For example I cannot use mPDF with newest version of iio/libmergepdf.

### PR DESCRIPTION
**Pull request by JanSlabon**
# BC breaks

- Removed SetImportUse() method.
- Signature of importPage() changed to default signature of FPDI (related to #480). Currently no alternative is implemented.

This fixes #578, #608 

Please review and comment. I am out of the office until 5th November.

### CODE REVIEW

File: src/FpdiTrait.php [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	 * A counter for template ids.
+	 *
+	 * @var int
+	 */
+	protected $templateId = 0;
+
+	protected function setPageFormat($format, $orientation)
+	{
+		// in mPDF this needs to be "P" (why ever)
+		$orientation = 'P';
```
Comment (JanSlabon) [inline src/FpdiTrait.php]: This is a very strange behavior of mPDF and I didn't found any other reasonable way to "fix" this.

File: src/Writer/ResourceWriter.php [hunk 2]
```diff
@@ -174,38 +164,14 @@ public function writeResources() // _putresources
...
-								foreach ($v[1] as $k2 => $v2) {
-									$this->writer->write($k2 . ' ', false);
-									$this->mpdf->pdf_write_value($v2);
-								}
-							}
-						}
-					}
-				}
-			}
-
```
Comment (JanSlabon) [inline src/Writer/ResourceWriter.php]: We never received any report for FPDI regarding such behavior/issue. Maybe it was an issue in a viewer application but from the technical view this is simply not required.

**Comment by JanSlabon** [comment]
Would it make sense to re-introduce the fault signature of importPage() mentioned in #480 to get this into the 7.x version? I mean this could be possible but at least I have no idea to write a test for it as it's simply wrong :-) 

Is there a timing for version 8?

**Comment by JanSlabon** [comment]
Exactly. FPDI v1 cannot be used with this pull request.

**Comment by rostockahoi** [comment]
Hi, I have trouble processing some PDFs with mPDF 7.1. I get an exception which guides me to https://www.setasign.com/products/fpdi-pdf-parser/details.

```
This document (/data/www/html/**/qo1qhe.pdf) probably uses a compression technique which is not supported by the free parser shipped with FPDI. (See https://www.setasign.com/fpdi-pdf-parser for more details)
```

Could this be solved by updating to FPDI 2 and using the commercial FPDI parser?

Thank you.

**Comment by finwe** [comment]
@rostockahoi Please do not abuse this conversion for not-entirely-related general questions.

**Comment by finwe** [comment]
@JanSlabon This looks OK. Can you also please look at examples repo? The old ImportPage signature is used and you'll certainly be faster in changing the calls to end in a same result as before.

**Comment by JanSlabon** [comment]
Done: https://github.com/mpdf/mpdf-examples/pull/20

**Comment by finwe** [comment]
Awesome, thanks.

**Pull request closed without merging**